src/Native/System.Device.Gpio.Native/libgpiod.c

```diff
@@ -0,0 +1,174 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <gpiod.h>
+
+/**
+ * @brief Close a GPIO chip handle and release all allocated resources.
+ * @param chip The GPIO chip object.
+ */
+extern void CloseChip(struct gpiod_chip *chip) 
+{
+	gpiod_chip_close(chip);
+}
+
+/**
+ * @brief Get the number of GPIO lines exposed by this chip.
+ * @param chip The GPIO chip object.
+ * @return Number of GPIO lines.
+ */
+extern unsigned int GetNumberOfLines(struct gpiod_chip *chip) 
+{
+	return gpiod_chip_num_lines(chip);
+}
+
+/**
+ * @brief Get the handle to the GPIO line at given offset.
+ * @param chip The GPIO chip object.
+ * @param offset The offset of the GPIO line.
+ * @return Pointer to the GPIO line handle or NULL if an error occured.
+ */
+extern struct gpiod_line *GetChipLineByOffset(struct gpiod_chip *chip, unsigned int offset)
+{
+	return gpiod_chip_get_line(chip, offset);
+}
+
+/**
+ * @brief Read the GPIO line direction setting.
+ * @param line GPIO line object.
+ * @return Returns GPIOD_DIRECTION_INPUT or GPIOD_DIRECTION_OUTPUT.
+ */
+extern int GetLineDirection(struct gpiod_line *line) 
+{
+	return gpiod_line_direction(line);
+}
+
+/**
+ * @brief Reserve a single line, set the direction to input.
+ * @param line GPIO line object.
+ * @param consumer Name of the consumer.
+ * @return 0 if the line was properly reserved, -1 on failure.
+ */
+extern int RequestLineInput(struct gpiod_line *line, const char *consumer) 
+{
+	return gpiod_line_request_input(line, consumer);
+}
+
+/**
+ * @brief Reserve a single line, set the direction to output.
+ * @param line GPIO line object.
+ * @param consumer Name of the consumer.
+ * @return 0 if the line was properly reserved, -1 on failure.
+ */
+extern int RequestLineOutput(struct gpiod_line *line, const char *consumer) 
+{
+	return gpiod_line_request_output(line, consumer, 0);
+}
+
+/**
+ * @brief Request all event type notifications on a single line.
+ * @param line GPIO line object.
+ * @param consumer Name of the consumer.
+ * @return 0 if the operation succeeds, -1 on failure.
+ */
+extern int RequestBothEdgeEventForLine(struct gpiod_line *line, const char *consumer) 
```

Comment:
NIT: Should we keep the Eventing related c APIs for the eventing review instead?

Reply:
we can remove that too, maybe that is better for future review

Reply:
i think this is fine